reframe/core/warnings.py

```diff
@@ -0,0 +1,58 @@
+import inspect
+import warnings
+
+
+class ReframeDeprecationWarning(DeprecationWarning):
+    '''Warning raised for deprecated features of the framework.'''
+
+
+warnings.filterwarnings('default', category=ReframeDeprecationWarning)
+
+
+_format_warning_orig = warnings.formatwarning
+
+
+def _format_warning(message, category, filename, lineno, line=None):
+    import reframe.core.runtime as rt
+    import reframe.utility.color as color
+
+    if category != ReframeDeprecationWarning:
+        return _format_warning_orig
+
+    if line is None:
+        # Read in the line from the file
+        with open(filename) as fp:
+            try:
+                line = fp.readlines()[lineno-1]
+            except IndexError:
+                line = '<no line information>'
+
+    message = f'{filename}:{lineno}: WARNING: {message}\n{line}\n'
+    if rt.runtime().get_option('general/0/colorize'):
```

Comment:
This needs a check whether a runtime is initialized. For example when trying to build the docs now, it fails because no runtime is unitialized.

Reply:
Hmm, interesting. I'll have a look!